### ISSUE TITLE
Auth fail on latest Hikvision (ONVIF 19.x)

### ISSUE THREAD

**Issue by cedricve** [IOTechSystems/onvif#41]
Thanks for your hard work on this and keeping the original onvif library a life! It looks like you have added a digest http client on top, well done! Currently I'm having issues with some Hikvision running ONVIF 19, the camera throws following errors, on using the digest authentication.
    
    [12-16 21:07:49][pid:0][ONVIF][ERROR]=====ABG, ret_stat:1
    appweb: 1: The url does not match the nonce url.
    
    [12-16 21:07:51][pid:0][ONVIF][ERROR]web_auth is MPR_ERR_DIGEST_AUTH_FAIL!

On older Hikvision running ONVIF 17, it does work properly.
Any thoughts?

**Comment by cloudxxx8** [comment]
You may ask the camera provider.

**Comment by jiekechoo** [comment]
@cedricve @cloudxxx8 When I tring connect to Hikvsion IPC (version 19.*), can't use digest authentication. I found most IPCs have SHA256 signature, but the onvif respository still using SHA1, maybe it is the issue.

**Comment by cedricve** [comment]
Yes indeed I will check that out, we have made our own fork here: https://github.com/kerberos-io/onvif

Keep you posted @jiekechoo

**Comment by cloudxxx8** [comment]
@cedricve could you please create a PR? then, we may share to a larger community.

**Comment by cedricve** [comment]
Will verify if this would fix the issue. Thanks for sharing @jiekechoo
